hardware/chisel/src/main/scala/core/Compute.scala

```diff
@@ -118,44 +123,102 @@ class Compute(debug: Boolean = false)(implicit p: Parameters) extends Module {
-  tensorGemm.io.out.rd.data.valid := io.out.rd.data.valid & dec.io.isGemm
-  tensorGemm.io.out.rd.data.bits <> io.out.rd.data.bits
+  for (idx <- 0 until tensorGemm.io.acc.splitWidth) {
+    tensorGemm.io.acc.rd(idx).data.valid :=
+      tensorAcc.io.tensor.rd(idx).data.valid & RegNext(dec.io.isGemm, init = false.B)
+    tensorGemm.io.acc.rd(idx).data.bits <> tensorAcc.io.tensor.rd(idx).data.bits
+  }
+  for (idx <- 0 until tensorGemm.io.out.splitWidth) {
+    tensorGemm.io.out.rd(idx).data.valid :=
+      io.out.rd(idx).data.valid & RegNext(dec.io.isGemm, init = false.B)
+    tensorGemm.io.out.rd(idx).data.bits <> io.out.rd(idx).data.bits
+  }
 
   // alu
-  tensorAlu.io.start := state === sIdle & start & dec.io.isAlu
-  tensorAlu.io.inst := inst_q.io.deq.bits
+  tensorAlu.io.start := RegNext(state === sIdle & start & dec.io.isAlu, init = false.B)
+  tensorAlu.io.dec := inst_q.io.deq.bits.asTypeOf(new AluDecode)
   tensorAlu.io.uop.data.valid := loadUop.io.uop.data.valid & dec.io.isAlu
   tensorAlu.io.uop.data.bits <> loadUop.io.uop.data.bits
-  tensorAlu.io.acc.rd.data.valid := tensorAcc.io.tensor.rd.data.valid & dec.io.isAlu
-  tensorAlu.io.acc.rd.data.bits <> tensorAcc.io.tensor.rd.data.bits
-  tensorAlu.io.out.rd.data.valid := io.out.rd.data.valid & dec.io.isAlu
-  tensorAlu.io.out.rd.data.bits <> io.out.rd.data.bits
+  for (idx <- 0 until tensorAlu.io.acc.splitWidth) {
+    tensorAlu.io.acc.rd(idx).data.valid :=
+      tensorAcc.io.tensor.rd(idx).data.valid & RegNext(dec.io.isAlu, init = false.B)
+    tensorAlu.io.acc.rd(idx).data.bits <> tensorAcc.io.tensor.rd(idx).data.bits
+  }
+  for (idx <- 0 until tensorAlu.io.out.splitWidth) {
+    tensorAlu.io.out.rd(idx).data.valid :=
+      io.out.rd(idx).data.valid & RegNext(dec.io.isAlu, init = false.B)
+    tensorAlu.io.out.rd(idx).data.bits <> io.out.rd(idx).data.bits
+  }
 
   // out
-  io.out.rd.idx <> Mux(dec.io.isGemm,
-    tensorGemm.io.out.rd.idx,
-    tensorAlu.io.out.rd.idx)
-  io.out.wr <> Mux(dec.io.isGemm, tensorGemm.io.out.wr, tensorAlu.io.out.wr)
+  for (idx <- 0 until tensorGemm.io.out.splitWidth) {
+    io.out.rd(idx).idx <> Mux(dec.io.isGemm,
+      tensorGemm.io.out.rd(idx).idx,
+      tensorAlu.io.out.rd(idx).idx)
+    assert( !tensorGemm.io.out.rd(idx).idx.valid || !tensorAlu.io.out.rd(idx).idx.valid)
+    assert( !tensorGemm.io.out.rd(idx).data.valid || !tensorAlu.io.out.rd(idx).data.valid)
 
+    assert( !tensorGemm.io.out.wr(idx).valid || !tensorAlu.io.out.wr(idx).valid)
+  }
+  require (tensorGemm.io.out.splitWidth == 1)
+  require (tensorAlu.io.out.splitWidth == 1)
+  io.out.wr(0).valid := Mux(
+    RegNext(dec.io.isGemm, init = false.B), tensorGemm.io.out.wr(0).valid, tensorAlu.io.out.wr(0).valid)
+  io.out.wr(0).bits.idx := Mux(
+    RegNext(dec.io.isGemm, init = false.B), tensorGemm.io.out.wr(0).bits.idx, tensorAlu.io.out.wr(0).bits.idx)
+  //put mux/Reg into every gemm group to build pipe (for Mux select) tree over distance
```

Comment:
could you add a space after the comment? like `// put` instead of `//put`

Reply:
All "//\S" occurrences replaced throughout PR